### DISCUSSION TITLE
get child parameters from rctx

### DISCUSSION THREAD

**Discussion by orhanerday** [99designs/gqlgen#2154] [Q&A] [answered]
I need to get the child parameter value from Graphql.

Here is my Graphql query:

     {
      foods(type: "fries") {
        drinks {
          calorie(joule :1500) {
            name
          }
        }
      }
    }

I can get the `type` field easily but also I need to get the `joule` value.
When I investigate this problem I find a file called `generated.go`, so I found the following code part for sending parameters to the target function;

	resTmp, err := ec.ResolverMiddleware(ctx, func(rctx context.Context) (interface{}, error) {
		ctx = rctx // use context from middleware stack in children
		return storage.FoodFinder(rctx, args["type"].(string)) //only args sended to function.
	})

I can get whole raw data from `rctx` variable but I don't wanna parse that, seems grueling. 

So How can I get `joule`, one of that child fields from `rctx` context variable?

Question Asked : [Stack Overflow](https://stackoverflow.com/questions/72185153/99designs-gqlgen-get-child-parameters-from-rctx)

**Comment by a8m** [comment, marked as answer]
There's an example for it in the integration tests (see [middleware_test.go](https://github.com/99designs/gqlgen/blob/master/codegen/testserver/singlefile/middleware_test.go#L67-L86)), but the main idea is as follows:

```go
opCtx := graphql.GetOperationContext(ctx)
collected := graphql.CollectFields(opCtx, fc.Field.Selections, []string{"User"})
child, err := fc.Child(ctx, collected[idx])
if err != nil {
    // ...
}
fmt.Println(*child.Args["arg"].(*int))
```
